imls-frontend/src/components/FetchData.vue

```diff
@@ -12,56 +13,105 @@ export default {
     path: {
       type: String,
       default: '/'
+    }, 
+    queryParams: {
+      type: Object,
+      default: () => {}
     }
   },
   data() {
     return {
-      state
+      store,
+      state,
+      fetchCount: null,
+      fetchError: {},
+      fetchedData: {},
+      isLoading: false,
     }
   },
   watch: {
-    'state.selectedDate'(newVal, oldVal) {
-      if (newVal !== oldVal) {
-        this.fetchDataFromState();
-      }
+    // this might not be the best way to react to selectedDate changes from the date picker. 
+    // However, we might refactor the date to use URL query params, which would make the
+    // query parameters available from the $route as props. 
+    // Vue would prefer to pass this down through props. Consider refactoring.
+    'state.selectedDate': {
+      async handler(newVal, oldVal)  {
+        console.log(newVal, oldVal)
+        if (newVal !== oldVal) {
+          await nextTick();
+          this.fetchData();
+        }
+      },
+      deep: true, 
     },
+    // maybe watch queryParams._start for now?
```

Comment:
remove comment?

Reply:
I believe this is related to work @sknep was doing.

Reply:
Yeah I have some work to do regarding watching front-end changes to the selected date. I don't really like how it's working today and wanted to remind myself that this is brittle and needs a refactor